source/PrintData.cpp

```diff
@@ -0,0 +1,473 @@
+/* PrintData.cpp
+Copyright (c) 2022 by warp-core
+
+Main function for Endless Sky, a space exploration and combat RPG.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "PrintData.h"
+
+#include "GameData.h"
+#include "Outfit.h"
+#include "Ship.h"
+
+#include <iostream>
+#include <map>
+#include <set>
+
+using namespace std;
+
+
+
+void PrintData::Print(char *argv[])
+{
+	const char *const *it = argv + 2;
+	if(!*it)
+		return;
+	string arg = *it;
+	if(arg == "-s" || arg == "--ships")
+		Ships(argv);
+	else if(arg == "-w" || arg == "--weapons")
+		Weapons(argv);
+	else if(arg == "-e" || arg == "--engines")
+		PrintEngineStats();
+	else if(arg == "-p" || arg == "--power")
+		PrintPowerStats();
+	else if(arg == "-o" || arg == "--outfits")
+		Outfits(argv);
+	else if(arg == "-h" || arg == "--help")
+		Help();
+	cout.flush();
+}
+
+
+
+void PrintData::Help()
+{
+	cerr << "--printdata: follow with arguments to print diffferent data." << endl;
```

Comment:
I don't think we should have `--printdata`. Simply using `--ships` or `--weapons` seems fine to me.

Reply:
The syntax is:
`EndlessSky.exe --printdata <arguments for specific data you want>`
Skipping `--printdata` would mean the logic for processing the specific input would need to be handled in main.cpp, which I wanted to avoid.

Reply:
I agree with quyykk on this one.